### PR TITLE
cavalier: 2023.01.29 -> 2024.1.0, refactor

### PR DESCRIPTION
**Pull request by zendo**
## Description of changes

https://github.com/NickvisionApps/Cavalier

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by zendo** [comment]
related https://github.com/NixOS/nixpkgs/issues/285701

**Comment by Steinhagen** [comment]
Haven't managed to get this to work:

```
✗  nix run github:zendo/nixpkgs/upd/cavalier#cavalier  

(process:19445): Adwaita-WARNING **: 17:51:47.077: Using GtkSettings:gtk-application-prefer-dark-theme with libadwaita is unsupported. Please use AdwStyleManager:color-scheme instead.
[AURA] Listening to commands.
UnhandledException - unhandled exception: System.ComponentModel.Win32Exception (2): An error occurred trying to start process 'cava' with working directory '/home/username/nixpkgs'. No such file or directory
   at System.Diagnostics.Process.ForkAndExecProcess(ProcessStartInfo startInfo, String resolvedFilename, String[] argv, String[] envp, String cwd, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at NickvisionCavalier.Shared.Models.CAVA.Restart() in /build/source/NickvisionCavalier.Shared/Models/CAVA.cs:line 99
   at NickvisionCavalier.Shared.Controllers.DrawingViewController..ctor() in /build/source/NickvisionCavalier.Shared/Controllers/DrawingViewController.cs:line 23
   at NickvisionCavalier.GNOME.Views.MainWindow..ctor(Builder builder, MainWindowController controller, Application application) in /build/source/NickvisionCavalier.GNOME/Views/MainWindow.cs:line 46
   at NickvisionCavalier.GNOME.Views.MainWindow..ctor(MainWindowController controller, Application application) in /build/source/NickvisionCavalier.GNOME/Views/MainWindow.cs:line 128
   at NickvisionCavalier.GNOME.Program.OnActivate(Application sedner, EventArgs e) in /build/source/NickvisionCavalier.GNOME/Program.cs:line 105
   at GObject.Signal`1.<>c__DisplayClass11_0.<Connect>b__0(Value returnValue, Value[] parameters)
   at GObject.Closure.InternalCallback(IntPtr closure, IntPtr returnValuePtr, UInt32 nParamValues, ValueData[] paramValuesData, IntPtr invocationHint, IntPtr userData)

```

**Comment by zendo** [comment]
It should be fixed.

**Comment by Steinhagen** [comment]
Works as expected and code looks good!

**Review by Ligthiago — APPROVED** [review]
Result of `nixpkgs-review pr 285967` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cavalier</li>
  </ul>
</details>

Tested basic functionality, works much better than older version that was almost unusable for me.

File: pkgs/by-name/ca/cavalier/package.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  runtimeDeps = [
+    glib
+    gtk4
+    libadwaita
+    librsvg
+  ];
+
+  postInstall = ''
+    substituteInPlace NickvisionCavalier.Shared/Linux/org.nickvision.cavalier.desktop.in --replace '@EXEC@' "NickvisionCavalier.GNOME"
```
Comment (SuperSandro2000) [inline pkgs/by-name/ca/cavalier/package.nix:57]:
```suggestion
    substituteInPlace NickvisionCavalier.Shared/Linux/org.nickvision.cavalier.desktop.in  \
      --replace-fail '@EXEC@' "NickvisionCavalier.GNOME"
```

File: pkgs/by-name/ca/cavalier/package.nix [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    desktop-file-utils
+  ];
+
+  buildInputs = [
+    glib
+    gtk4
+    libadwaita
+  ];
+
+  runtimeDeps = [
```
Comment (SuperSandro2000) [inline pkgs/by-name/ca/cavalier/package.nix:49]: Do we really need those in LD_LIBRARY_PATH or does it work without them?

**Pull request merged**